lib/base-package-manager.ts

```diff
@@ -0,0 +1,94 @@
+import * as child_process from 'child_process';
+
+export class BasePackageManager {
+	constructor(private packageManager: string) { }
+
+	protected getNpmExecutableName(isWindows: boolean): string {
+		let npmExecutableName = this.packageManager;
+
+		if (isWindows) {
+			npmExecutableName += ".cmd";
+		}
+
+		return npmExecutableName;
+	}
+
+	protected async processPackageManagerInstall(
+		childProcess: child_process.ChildProcess,
+		isWindows: boolean,
+		params: string[],
+	): Promise<ISpawnResult> {
+		return new Promise<ISpawnResult>((resolve, reject) => {
+			let isFulfilled = false;
+			let capturedOut = "";
+			let capturedErr = "";
+
+			const npmExecutable = this.getNpmExecutableName(isWindows);
+
+			if (childProcess.stdout) {
```

Comment:
This code is duplicated to this https://github.com/NativeScript/nativescript-cli/blob/master/lib/common/child-process.ts#L57
So I suggest to inject `$childProcess: IChildProcess` in constructor and reuse the logic here.
```
protected async processPackageManagerInstall(params: string[], opts: { cwd: string }) {
    const npmExecutable = this.getNpmExecutableName();
    const stdioValue = isInteractive() ? "inherit" : "pipe";
    const result = await this.$childProcess.spawnFromEvent(npmExecutable, params, "close", { cwd: opts.cwd, stdio: stdioValue });
    return result;
}
```